CMakeLists.txt

```diff
@@ -13,20 +16,27 @@ set(CMAKE_CXX_STANDARD 11)
 # Run Autocoder when CMake is run
 execute_process(COMMAND bash ${PROJECT_SOURCE_DIR}/bloop-dev autocode)
 
+if(DEFINED STUB_BUS)
+    add_definitions(-DSTUB_BUS=${STUB_BUS})
+endif()
 
-IF(TEST)
-    ADD_DEFINITIONS(-DTEST)
-    set(dependecies tests.cpp)
-ELSE()
-     set(dependecies main.cpp)
-ENDIF()
+if(DEFINED DEV)
+    add_definitions(-DDEV)
+    set(dependencies tests.cpp)
+else()
+    set(dependencies main.cpp)
+endif()
 
 add_subdirectory(embedded/devices)
 add_subdirectory(embedded/drivers)
-set(libraries devices
-    drivers)
+add_subdirectory(btf)
+
+set(libraries
+    devices
+    drivers
+    pthread)
 
-add_executable(badgerloop ${dependecies})
+add_executable(badgerloop ${dependencies})

```

Comment:
when building test stuff we should make a different executable name, `badgerloop_test`